# Airline Passenger Satisfaction


This dataset is given by an airline organization, but it's actual name is not given due to various purposes. 
I use this dataset to create dashboard to uncover insights in improving customer experience and operational efficiency at Airline X.

Before using this dataset to create dashboard, I conducted data investigation and data cleaning so that the data which will be visualized represent the actual condition.

The source of this dataset is: https://www.kaggle.com/datasets/sjleshrac/airlines-customer-satisfaction/data

In [1]:
# import libraries
import numpy as np
import pandas as pd

In [2]:
# load the dataset
data = pd.read_csv(r"E:\Infinite Case Study\airline_customer_satisfaction.csv")

In [3]:
# view the data
data.head()

satisfaction  Gender   Customer Type  Age   Type of Travel     Class  \
0    satisfied  Female  Loyal Customer   65  Personal Travel       Eco   
1    satisfied    Male  Loyal Customer   47  Personal Travel  Business   
2    satisfied  Female  Loyal Customer   15  Personal Travel       Eco   
3    satisfied  Female  Loyal Customer   60  Personal Travel       Eco   
4    satisfied  Female  Loyal Customer   70  Personal Travel       Eco   

   Flight Distance  Seat comfort  Departure/Arrival time convenient  \
0              265             0                                  0   
1             2464             0                                  0   
2             2138             0                                  0   
3              623             0                                  0   
4              354             0                                  0   

   Food and drink  ...  Online support  Ease of Online booking  \
0               0  ...               2                       3   
1               0  ...               2                       3   
2               0  ...               2                       2   
3               0  ...               3                       1   
4               0  ...               4                       2   

   On-board service  Leg room service  Baggage handling  Checkin service  \
0                 3                 0                 3                5   
1                 4                 4                 4                2   
2                 3                 3                 4                4   
3                 1                 0                 1                4   
4                 2                 0                 2                4   

   Cleanliness  Online boarding  Departure Delay in Minutes  \
0            3                2                           0   
1            3                2                         310   
2            4                2                           0   
3            1                3                           0   
4            2                5                           0   

   Arrival Delay in Minutes  
0                       0.0  
1                     305.0  
2                       0.0  
3                       0.0  
4                       0.0  

[5 rows x 23 columns]

In [4]:
# explore the size of the data
data.shape

(129880, 23)

There are 129880 rows and 23 columns in the dataset.

In [5]:
# check the data types
data.dtypes

satisfaction                          object
Gender                                object
Customer Type                         object
Age                                    int64
Type of Travel                        object
Class                                 object
Flight Distance                        int64
Seat comfort                           int64
Departure/Arrival time convenient      int64
Food and drink                         int64
Gate location                          int64
Inflight wifi service                  int64
Inflight entertainment                 int64
Online support                         int64
Ease of Online booking                 int64
On-board service                       int64
Leg room service                       int64
Baggage handling                       int64
Checkin service                        int64
Cleanliness                            int64
Online boarding                        int64
Departure Delay in Minutes             int64
Arrival De

In [6]:
# check for missing values
data.isna().sum()

satisfaction                           0
Gender                                 0
Customer Type                          0
Age                                    0
Type of Travel                         0
Class                                  0
Flight Distance                        0
Seat comfort                           0
Departure/Arrival time convenient      0
Food and drink                         0
Gate location                          0
Inflight wifi service                  0
Inflight entertainment                 0
Online support                         0
Ease of Online booking                 0
On-board service                       0
Leg room service                       0
Baggage handling                       0
Checkin service                        0
Cleanliness                            0
Online boarding                        0
Departure Delay in Minutes             0
Arrival Delay in Minutes             393
dtype: int64

There are only 393 missing values out of the 129880 values. Since this is a tiny percentage of the total, I will proceed to drop these rows.

In [7]:
# delete missing values
data.dropna(axis=0).reset_index(drop=True)

satisfaction  Gender      Customer Type  Age   Type of Travel  \
0          satisfied  Female     Loyal Customer   65  Personal Travel   
1          satisfied    Male     Loyal Customer   47  Personal Travel   
2          satisfied  Female     Loyal Customer   15  Personal Travel   
3          satisfied  Female     Loyal Customer   60  Personal Travel   
4          satisfied  Female     Loyal Customer   70  Personal Travel   
...              ...     ...                ...  ...              ...   
129482     satisfied  Female  disloyal Customer   29  Personal Travel   
129483  dissatisfied    Male  disloyal Customer   63  Personal Travel   
129484  dissatisfied    Male  disloyal Customer   69  Personal Travel   
129485  dissatisfied    Male  disloyal Customer   66  Personal Travel   
129486  dissatisfied  Female  disloyal Customer   38  Personal Travel   

           Class  Flight Distance  Seat comfort  \
0            Eco              265             0   
1       Business             2464             0   
2            Eco             2138             0   
3            Eco              623             0   
4            Eco              354             0   
...          ...              ...           ...   
129482       Eco             1731             5   
129483  Business             2087             2   
129484       Eco             2320             3   
129485       Eco             2450             3   
129486       Eco             4307             3   

        Departure/Arrival time convenient  Food and drink  ...  \
0                                       0               0  ...   
1                                       0               0  ...   
2                                       0               0  ...   
3                                       0               0  ...   
4                                       0               0  ...   
...                                   ...             ...  ...   
129482                                  5               5  ...   
129483                                  3               2  ...   
129484                                  0               3  ...   
129485                                  2               3  ...   
129486                                  4               3  ...   

        Online support  Ease of Online booking  On-board service  \
0                    2                       3                 3   
1                    2                       3                 4   
2                    2                       2                 3   
3                    3                       1                 1   
4                    4                       2                 2   
...                ...                     ...               ...   
129482               2                       2                 3   
129483               1                       3                 2   
129484               2                       4                 4   
129485               2                       3                 3   
129486               3                       4                 5   

        Leg room service  Baggage handling  Checkin service  Cleanliness  \
0                      0                 3                5            3   
1                      4                 4                2            3   
2                      3                 4                4            4   
3                      0                 1                4            1   
4                      0                 2                4            2   
...                  ...               ...              ...          ...   
129482                 3                 4                4            4   
129483                 3                 3                1            2   
129484                 3                 4                2            3   
129485                 2                 3                2            1   
129486                 5                 5                3            3

In [8]:
# investigate descriptive statistics
data.describe()

Age  Flight Distance   Seat comfort  \
count  129880.000000    129880.000000  129880.000000   
mean       39.427957      1981.409055       2.838597   
std        15.119360      1027.115606       1.392983   
min         7.000000        50.000000       0.000000   
25%        27.000000      1359.000000       2.000000   
50%        40.000000      1925.000000       3.000000   
75%        51.000000      2544.000000       4.000000   
max        85.000000      6951.000000       5.000000   

       Departure/Arrival time convenient  Food and drink  Gate location  \
count                      129880.000000   129880.000000  129880.000000   
mean                            2.990645        2.851994       2.990422   
std                             1.527224        1.443729       1.305970   
min                             0.000000        0.000000       0.000000   
25%                             2.000000        2.000000       2.000000   
50%                             3.000000        3.000000       3.000000   
75%                             4.000000        4.000000       4.000000   
max                             5.000000        5.000000       5.000000   

       Inflight wifi service  Inflight entertainment  Online support  \
count          129880.000000           129880.000000   129880.000000   
mean                3.249130                3.383477        3.519703   
std                 1.318818                1.346059        1.306511   
min                 0.000000                0.000000        0.000000   
25%                 2.000000                2.000000        3.000000   
50%                 3.000000                4.000000        4.000000   
75%                 4.000000                4.000000        5.000000   
max                 5.000000                5.000000        5.000000   

       Ease of Online booking  On-board service  Leg room service  \
count           129880.000000     129880.000000     129880.000000   
mean                 3.472105          3.465075          3.485902   
std                  1.305560          1.270836          1.292226   
min                  0.000000          0.000000          0.000000   
25%                  2.000000          3.000000          2.000000   
50%                  4.000000          4.000000          4.000000   
75%                  5.000000          4.000000          5.000000   
max                  5.000000          5.000000          5.000000   

       Baggage handling  Checkin service    Cleanliness  Online boarding  \
count     129880.000000    129880.000000  129880.000000    129880.000000   
mean           3.695673         3.340807       3.705759         3.352587   
std            1.156483         1.260582       1.151774         1.298715   
min            1.000000         0.000000       0.000000         0.000000   
25%            3.000000         3.000000       3.000000         2.000000   
50%            4.000000         3.000000       4.000000         4.000000   
75%            5.000000         4.000000       5.000000         4.000000   
max            5.000000         5.000000       5.000000         5.000000   

       Departure Delay in Minutes  Arrival Delay in Minutes  
count               129880.000000             129487.000000  
mean                    14.713713                 15.091129  
std                     38.071126                 38.465650  
min                      0.000000                  0.000000  
25%                      0.000000                  0.000000  
50%                      0.000000                  0.000000  
75%                     12.000000                 13.000000  
max                   1592.000000               1584.000000

The descriptive statistics seems nice, there is no outlier in most of the data. We can proceed to use this data for dashboard visualization.

In [10]:
# save the cleaned_data as csv file
cleaned_data = data.to_csv(r"E:\Infinite Case Study\cleaned_data_airline.csv", index=False)